go-controller/pkg/node/node.go

```diff
@@ -250,10 +252,45 @@ func (n *OvnNode) Start() error {
 	if !ok {
 		return fmt.Errorf("Cannot get kubeclient for starting CNI server")
 	}
+	err = n.WatchEndpoints()
+	if err != nil {
+		return err
+	}
 
 	// start the cni server
 	cniServer := cni.NewCNIServer("", kclient.KClient)
 	err = cniServer.Start(cni.HandleCNIRequest)
 
 	return err
 }
+
+func (n *OvnNode) WatchEndpoints() error {
+	_, err := n.watchFactory.AddEndpointsHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, new interface{}) {
+			epNew := new.(*kapi.Endpoints)
+			epOld := old.(*kapi.Endpoints)
+			if reflect.DeepEqual(epNew.Subsets, epOld.Subsets) {
+				return
+			}
+			newEndpointAddressMap := buildEndpointAddressMap(epNew.Subsets)
+			for _, subset := range epOld.Subsets {
+				for _, address := range subset.Addresses {
+					if _, ok := newEndpointAddressMap[address.IP]; ok {
+						deleteConntrack(address.IP)
```

Comment:
just thinking in loud, what about @dcbw comment of not deleting for TCP?
https://github.com/ovn-org/ovn-kubernetes/pull/1074#discussion_r385912010

Reply:
I don't think TCP will have the same problem but @JacobTanenbaum can you please test it and verify? With TCP this should happen:
1. Pod A exists as endpoint to a service (TCP connection is open from a client)
2. Pod A is deleted from a service and deleted as a pod (TCP close should be sent by pod, but maybe not if force termination)
3. Pod B comes up with same IP, Pod A continues to send TCP packet on same connection (assuming no close in #2)
4. Pod B will send TCP reset (packet will not be part of any existing connection)
5. Pod A sends TCP SYN, OVN selects new endpoint or sends TCP reset if no more endpoints exist

Note also the TCP timeout is smaller than UDP (10 sec vs 30):
nf_conntrack_tcp_timeout_close - INTEGER (seconds)
	default 10
and TCP loose should be disabled:
nf_conntrack_tcp_loose - BOOLEAN
	0 - disabled
	not 0 - enabled (default)

	If it is set to zero, we disable picking up already established
	connections.

Reply:
TCP does not have this problem, TCP selects another endpoint or if it is the only endpoint closes the connection when the pod gets deleted